test/static_test.rb

```diff
@@ -59,6 +59,12 @@ class StaticTest < Minitest::Test
     assert not_found?
   end
 
+  it 'passes to the next handler when the path contains null bytes' do
+    @app.set :public_folder, "\0"
+    get "/"
+    assert not_found?
```

Comment:
I don't think this test covers the case correctly, because [the `:static` is set to `true`](https://github.com/sinatra/sinatra/blob/a4dd24add24f2dd0e7299b9e68e12038138294d3/test/static_test.rb#L6).
If you don't set `:static` and [use the default settings](https://github.com/sinatra/sinatra/blob/a4dd24add24f2dd0e7299b9e68e12038138294d3/lib/sinatra/base.rb#L1829), `File.exist?` will continue to raise an exception as you said. The `static?` is called in [dispatching](https://github.com/sinatra/sinatra/blob/a4dd24add24f2dd0e7299b9e68e12038138294d3/lib/sinatra/base.rb#L1096).

Reply:
Ah right. The request path is what we need to test, not the public folder. Can we change the test to look at a request path with a null byte?

```
get "/foo%00"
assert not_found?
````

I would expect the app to raise an exception if `public_folder` path did contain a null byte.

Reply:
@namusyaka & @jkowens that makes sense. I've amended the commit to reflect your feedback.

Reply:
@namusyaka & @jkowens bump for the message above 😁. How does this change look?

Reply:
@makushline looks good. I think a couple small tweaks and this should be ready to merge.